src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -258,39 +278,48 @@ private void HandleChannelException(IConnection connection, Exception ex)
                 return;
             }
 
-            if (ShouldInvalidateServer(ex))
+            if (connection.Generation != _connectionPool.Generation)
+            {
+                return; // stale generation number
+            }
+
+            if (ShouldInvalidateServer(connection, ex, Description, out TopologyVersion responseTopologyVersion))
```

Comment:
Using the property instead of the field is unusual and any future reader would be tempted to replace `Description` with `_description`. Use a temporary variable and comment to explain why the property must be used here:

    var description = Description; // use Description property to access _description value safely